### PR TITLE
[formrecognizer] re-enable tests

### PR DESCRIPTION
**Pull request by kristapratico**
Resolves https://github.com/Azure/azure-sdk-for-python/issues/20839

### CODE REVIEW

File: sdk/formrecognizer/azure-ai-formrecognizer/tests/test_receipt_async.py [hunk 1]
```diff
@@ -400,14 +400,13 @@ async def test_receipt_locale_specified(self, client):
...
     @GlobalClientPreparer()
-    @pytest.mark.skip("the service is returning a different error code")
     async def test_receipt_locale_error(self, client):
         with open(self.receipt_jpg, "rb") as fd:
             receipt = fd.read()
         with pytest.raises(HttpResponseError) as e:
             async with client:
                 await client.begin_analyze_document("prebuilt-receipt", receipt, locale="not a locale")
-        assert "UnsupportedLocale" == e.value.error.code
+        assert "InvalidArgument" == e.value.error.code
```
Comment (catalinaperalta) [inline sdk/formrecognizer/azure-ai-formrecognizer/tests/test_receipt_async.py:409]: So is this the error code to expect from now on here?

**Review by catalinaperalta — APPROVED** [review]
🚢

File: sdk/formrecognizer/azure-ai-formrecognizer/tests/test_receipt_async.py (hunk 1 above)
Comment (kristapratico) [reply sdk/formrecognizer/azure-ai-formrecognizer/tests/test_receipt_async.py:409]: Yes, for v2021-09-30-preview this is the error code. Also if the service changes up the error response due to REST feedback during the betas we'll know with this test.

**Pull request merged**